File: Python/random_forest_regression.py
# -*- coding: utf-8 -*-
"""
ML Lab for the udemy course "Machine Learning A-Z (Codes and Datasets)"
Topic: Random Forest Regression
@author: Anand
"""

import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
from sklearn.ensemble import RandomForestRegressor
dataset = pd.read_csv("../Data/Position_Salaries.csv")

X = dataset.iloc[:,1:-1]
y = dataset.iloc[:,-1]

#Creating the Ranodom Forest Model
rfr = RandomForestRegressor(n_estimators=10, random_state=0)
rfr.fit(X, y)

#Predicting using the model
x_test = [[6.5]]
y_pred = rfr.predict(x_test)

#PLotting the data(hi res)
X_grid = np.arange(min(X.values), max(X.values), 0.01)
X_grid = X_grid.reshape(len(X_grid), 1)
plt.scatter(X, y, color = "red")
plt.plot(X_grid, rfr.predict(X_grid), color = "blue")
plt.title("Random Forest Regression")
plt.xlabel("Years of Experience")
plt.ylabel("Salary")
plt.show()

File: R/random_forest_regression.R
#' 
#' ML Lab for the udemy course "Machine Learning A-Z (Codes and Datasets)"
#' Topic: Random Forest Regression
#' @author: Anand

#Import the dataset#
dataset <- read.csv("../Data/Position_Salaries.csv")
dataset <- dataset[2:3]

#Training the model
set.seed(1234)
library(randomForest)
regressor = randomForest(x = dataset[1], y = dataset$Salary,ntree = 500)
y_pred <- predict(regressor, newdata = data.frame(Level = 6.5))

#Visualising the results#
library(ggplot2)
x_grid =seq(min(dataset$Level), max(dataset$Level), 0.01)
ggplot() +
  geom_point(aes(x = dataset$Level, y = dataset$Salary), colour = "red") +
  geom_line(aes(x = x_grid, y = predict(regressor, newdata = data.frame(Level = x_grid))), colour = "blue") +
  ggtitle("Sal Vs Level") +
  xlab("Years of Experience") +
  ylab("Salary")

